src/iosrv/srv.c

```diff
@@ -1199,6 +1199,21 @@ dss_collective_reduce_internal(struct dss_coll_ops *ops,
 		stream			= &stream_args->csa_streams[tid];
 		stream->st_coll_args	= &carg;
 
+		if (args->ca_exclude_tgts_cnt) {
+			int i;
+
+			for (i = 0; i < args->ca_exclude_tgts_cnt; i++)
+				if (args->ca_exclude_tgts[i] == tid)
+					break;
+
+			if (i < args->ca_exclude_tgts_cnt) {
+				D_DEBUG(DB_TRACE, "Skip tgt %d\n", tid);
+				rc = ABT_future_set(future, (void *)stream);
+				D_ASSERTF(rc == ABT_SUCCESS, "%d\n", rc);
+				continue;
+			}
+		}
```

Comment:
I'm wondering if it's a good idea to add this feature to dss_collective call. dss_collective call is just a generic function to  execute a collective_func() over all target xstreams, is it better to just do the target status check in each individual specific collective_func() (not every kind of collective call need to check target status)?